docs/src/literate/ns_vs_diffeq.jl

```diff
@@ -0,0 +1,463 @@
+    push!(hole_face_ring, FaceIndex((hci+x_cells, 1)))
+end
+grid.facesets["hole"] = Set(filter(x->x.idx[1] ∉ hole_cell_indices, collect(hole_face_ring)));
+cell_indices_map = map(ci->norm(mean(map(i->grid.nodes[i].x-[0.2,0.2], Ferrite.vertices(grid.cells[ci]))))>0.05 ? indexin([ci], cell_indices)[1] : 0, 1:length(grid.cells))
+grid.cells = grid.cells[cell_indices]
+for facesetname in keys(grid.facesets)
+    grid.facesets[facesetname] = Set(map(fi -> FaceIndex( cell_indices_map[fi.idx[1]] ,fi.idx[2]), collect(grid.facesets[facesetname])))
+end;
+
+# We test against full development of the flow - so regenerate the grid                              #src
+grid = generate_grid(Quadrilateral, (x_cells, y_cells), Vec{2}((0.0, 0.0)), Vec{2}((0.55, 0.41)));   #hide
+
+# ### Function space
+# To ensure stability we utilize the Taylor-Hood element pair Q2-Q1.
+# We have to utilize the same quadrature rule because in the weak form the
+# linear pressure term is tested against a quadratic function.
+ip_v = Lagrange{dim, RefCube, 2}()
+ip_geom = Lagrange{dim, RefCube, 1}()
+qr = QuadratureRule{dim, RefCube}(4)
+cellvalues_v = CellVectorValues(qr, ip_v, ip_geom);
+
+ip_p = Lagrange{dim, RefCube, 1}()
+cellvalues_p = CellScalarValues(qr, ip_p, ip_geom);
+
+dh = DofHandler(grid)
+push!(dh, :v, dim, ip_v)
+push!(dh, :p, 1, ip_p)
+close!(dh);
+
+# ### Boundary conditions
+# As in the DFG benchmark we apply no-slip conditions to the top, bottom and
+# cylinder boundary. The no-slip condition states that the velocity of the
+# fluid on this portion of the boundary is fixed to be zero.
+ch = ConstraintHandler(dh);
+
+nosplip_face_names = ["top", "bottom", "hole"];
+# No hole for the test present                                          #src
+nosplip_face_names = ["top", "bottom"]                                  #hide
+∂Ω_noslip = union(getfaceset.((grid, ), nosplip_face_names)...);
+noslip_bc = Dirichlet(:v, ∂Ω_noslip, (x, t) -> [0,0], [1,2])
+add!(ch, noslip_bc);
+
+# The left boundary has a parabolic inflow with peak velocity of 1.0. This
+# ensures that for the given geometry the Reynolds number is 100, which
+# is already enough to obtain some simple vortex streets. By increasing the
+# velocity further we can obtain stronger vortices - which may need additional
+# refinement of the grid.
+∂Ω_inflow = getfaceset(grid, "left");
+
+vᵢₙ(t) = clamp(t, 0.0, 1.0)*1.0 #inflow velocity
+vᵢₙ(t) = clamp(t, 0.0, 1.0)*0.3 #hide
+parabolic_inflow_profile((x,y),t) = [4*vᵢₙ(t)*y*(0.41-y)/0.41^2,0]
+inflow_bc = Dirichlet(:v, ∂Ω_inflow, parabolic_inflow_profile, [1,2])
+add!(ch, inflow_bc);
+
+# The outflow boundary condition has been applied on the right side of the
+# cylinder when the weak form has been derived by setting the boundary integral
+# to zero. It is also called the do-nothing condition. Other outflow conditions
+# are also possible.
+∂Ω_free = getfaceset(grid, "right");
+
+close!(ch)
+update!(ch, 0.0);
+
+# ### Linear System Assembly
```

Comment:
maybe add here once again the weak form and mark with `underbrace` which parts correspond to stokes, advective and mass terms

Reply:
Is this also resolved?

Reply:
yes :)